django/test/runner.py

```diff
@@ -264,10 +286,22 @@ def addExpectedFailure(self, test, err):
             err = err[0], err[1], None
         self.check_picklable(test, err)
         self.events.append(('addExpectedFailure', self.test_index, err))
+        super().addExpectedFailure(test, err)
 
     def addUnexpectedSuccess(self, test):
         self.events.append(('addUnexpectedSuccess', self.test_index))
-        self.stop_if_failfast()
+        super().addUnexpectedSuccess(test)
+
+    def wasSuccessful(self):
+        """Tells whether or not this result was a success."""
+        failure_types = {'addError', 'addFailure', 'addUnexpectedSuccess'}
```

Comment:
I think we should add `addSubTest` to this set. I added an assertion that fails.

Reply:
Yup, well spotted!